code/modules/mob/living/simple_animal/hostile/mining/elites/herald.dm

```diff
@@ -305,9 +305,9 @@
 	usr.visible_message("<span class='warning'>[usr] starts to crawl into [starting_mirror]...</span>", \
 			"<span class='notice'>You start to crawl into the [starting_mirror]...</span>")
 	if(do_after(usr, 2 SECONDS, target = usr))
-		if(QDELETED(chosen) || !usr|| usr.incapacitated() || !chosen || (get_dist(src, starting_mirror) > 1))
-			return
 		var/turf/destination = get_turf(chosen)
+		if(QDELETED(chosen) || !usr|| usr.incapacitated() || !chosen || (get_dist(src, starting_mirror) > 1 || destination.z != usr.z))
```

Comment:
check for TRAIT_HANDS_BLOCKED instead of `incapacitated` please

Reply:
done!